packages/block-editor/src/components/block-toolbar/utils.js

```diff
@@ -181,28 +175,13 @@ let requestAnimationFrameId;
 export function useToggleBlockHighlight( clientId ) {
 	const { toggleBlockHighlight } = useDispatch( 'core/block-editor' );
 
-	const updateBlockHighlight = useCallback(
-		( isFocused ) => {
-			toggleBlockHighlight( clientId, isFocused );
-		},
-		[ clientId ]
-	);
+	const updateBlockHighlight = ( isFocused ) => {
+		toggleBlockHighlight( clientId, isFocused );
+	};
 
 	useEffect( () => {
-		// On mount, we make sure to cancel any pending animation frame request
-		// that hasn't been completed yet. Components like NavigableToolbar may
-		// mount and unmount quickly.
-		if ( requestAnimationFrameId ) {
-			cancelAnimationFrame( requestAnimationFrameId );
-		}
-		return () => {
-			// Sequences state change to enable editor updates (e.g. cursor
-			// position) to render correctly.
-			requestAnimationFrameId = requestAnimationFrame( () => {
-				updateBlockHighlight( false );
-			} );
-		};
-	}, [] );
+		return () => toggleBlockHighlight( clientId, false );
```

Comment:
Why are we toggling block highlight in an effect rather than at the reducer level? For a SELECT_BLOCK action etc., the highlight state should change immediately. Cc @youknowriad.

Reply:
I agree that if we can do it automatically on the reducer, it's better and avoid an extra rerenders for all components.